pkgs/build-support/docker/default.nix

```diff
@@ -886,4 +888,70 @@ rec {
         done;
       '' + extraCommands;
     });
+
+  # This function builds a docker image that behaves like a nix-shell
+  buildNixShellImage =
+    # Use the environment of this derivation
+    { name
+    , drv
+    , tag ? null
+    # Extra packages available inside the image
+    , extraPackages ? []
```

Comment:
What about calling this `contents`? It seems to me it behaves like the `contents` attribute of our other docker tools.